addons/mail/models/mail_activity.py

```diff
@@ -110,6 +110,21 @@ def default_get(self, fields):
             res['res_model_id'] = self.env['ir.model']._get(res['res_model']).id
         return res
 
+    @api.model
+    def _default_activity_type_id(self):
+        ActivityType = self.env["mail.activity.type"]
+        model_name = self.default_get(['res_model'])['res_model']
```

Comment:
Actually the real field to use is res_model_id (res_model is only a computed based on it). So I would use that one to be sure :) .

You can also check the default after the return todo, no need to search for it if we don't use it (just ordering lines by use).